stdlib/3/_curses.pyi

```diff
@@ -292,227 +292,171 @@ def use_env(flag: bool) -> None: ...
-
     def attrset(self, attr: int) -> None: ...
-
-    def bkgd(self, ch: _chtype, attr: Optional[int]) -> None: ...
-
-    def bkgset(self, ch: _chtype, attr: Optional[int]) -> None: ...
-
-    def border(self, ls: Optional[int], rs: Optional[int],
-               ts: Optional[int], bs: Optional[int],
-               tl: Optional[int], tr: Optional[int],
-               bl: Optional[int], br: Optional[int]) -> None: ...
-
-    def box(self, vertch: Optional[Tuple[int, int]], horch: Optional[Tuple[int, int]]) -> None: ...
-
+    def bkgd(self, ch: _chtype, attr: int = ...) -> None: ...
+    def bkgset(self, ch: _chtype, attr: int = ...) -> None: ...
+    @overload
+    def border(self, ls: _chtype = ...) -> None: ...
+    @overload
+    def border(self, ls: _chtype = ..., rs: _chtype = ...) -> None: ...
+    @overload
+    def border(self, ls: _chtype = ..., rs: _chtype = ..., ts: _chtype = ...) -> None: ...
+    @overload
+    def border(self, ls: _chtype = ..., rs: _chtype = ..., ts: _chtype = ..., bs: _chtype = ...) -> None: ...
+    @overload
+    def border(self, ls: _chtype = ..., rs: _chtype = ..., ts: _chtype = ..., bs: _chtype = ..., tl: _chtype = ...) -> None: ...
+    @overload
+    def border(self, ls: _chtype = ..., rs: _chtype = ..., ts: _chtype = ..., bs: _chtype = ..., tl: _chtype = ..., tr: _chtype = ...) -> None: ...
+    @overload
+    def border(self, ls: _chtype = ..., rs: _chtype = ..., ts: _chtype = ..., bs: _chtype = ..., tl: _chtype = ..., tr: _chtype = ..., bl: _chtype = ...) -> None: ...
+    @overload
+    def border(self, ls: _chtype = ..., rs: _chtype = ..., ts: _chtype = ..., bs: _chtype = ..., tl: _chtype = ..., tr: _chtype = ..., bl: _chtype = ..., br: _chtype = ...) -> None: ...
+    @overload
+    def box(self) -> None: ...
+    @overload
+    def box(self, vertch: _chtype = ..., horch: _chtype = ...) -> None: ...
     @overload
     def chgat(self, attr: int) -> None: ...
-
     @overload
     def chgat(self, num: int, attr: int) -> None: ...
-
     @overload
     def chgat(self, y: int, x: int, attr: int) -> None: ...
-
     @overload
     def chgat(self, y: int, x: int, num: int, attr: int) -> None: ...
-
     def clear(self) -> None: ...
-
-    def clearok(self, yes: int): ...
-
+    def clearok(self, yes: int) -> None: ...
     def clrtobot(self) -> None: ...
-
     def clrtoeol(self) -> None: ...
-
     def cursyncup(self) -> None: ...
-
     @overload
     def delch(self) -> None: ...
-
     @overload
-    def delch(self, y: int, x: int) -> None: ...
-
+    def delch(self, y: int = ..., x: int = ...) -> None: ...
     def deleteln(self) -> None: ...
-
     @overload
     def derwin(self, begin_y: int, begin_x: int) -> '_CursesWindow': ...
-
     @overload
     def derwin(self, nlines: int, ncols: int, begin_y: int, begin_x: int) -> '_CursesWindow': ...
-
-    def echochar(self, ch: _chtype, attr: Optional[int]) -> None: ...
-
+    def echochar(self, ch: _chtype, attr: Optional[int] = ...) -> None: ...
```

Comment:
Is this really Optional (is None an allowed value for `attr`)?

Reply:
Considering the different experiments that I have done your are probably right. I initially made them all ```Optional[int] = ...``` and then realized my mistake and did the experiments to prove that they threw an exception when None was there. I may have missed this one in that process. I'll do the echochar experiment and get back to you.

Reply:
Exception thrown with a None in the second argument.